src/views/layout.njk

```diff
@@ -8,40 +8,53 @@
 {% block head %}
   <meta name="robots" content="noindex, nofollow" />
   <link rel="stylesheet" href="/eligibility-checker/stylesheets/main.css" />
-  <link rel="icon" type="image/svg+xml" href="/eligibility-checker/assets/images/favicon.svg">
+  <link rel="icon" type="image/svg+xml" href="/eligibility-checker/assets/images/favicon.svg" />
   <script>
     function handleSubmit(event, formId) {
       event.preventDefault();
-      const form = document.getElementById("questionForm");
+      const form = document.getElementById('questionForm');
       const formData = new FormData(form);
-      const url = "/eligibility-checker/{{ meta.grantTypeId }}/transition";
-      const data = { event: "NEXT", currentPageId: "{{ meta.currentPageId }}" , nextPageId: "{{ meta.nextPageId }}", previousPageId: "{{ meta.previousPageId }}", answer: formData.get("{{ meta.currentPageId }}") };
+      const url = '/eligibility-checker/{{ meta.grantTypeId }}/transition';
+      const data = {
+        event: 'NEXT',
+        currentPageId: '{{ meta.currentPageId }}',
+        nextPageId: '{{ meta.nextPageId }}',
+        previousPageId: '{{ meta.previousPageId }}',
+        questionType: '{{ meta.questionType }}',
+        answer: formData.get('{{ meta.currentPageId }}')
+      };
 
       fetch(url, {
-        method: "POST",
-        headers: { "Content-Type": "application/json" },
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
         body: JSON.stringify(data)
       })
-      .then(response => response.json())
-      .then(data => {
-        window.location.href = `/eligibility-checker/{{ meta.grantTypeId }}/${data.nextPageId}`;
-      })
-      .catch(error => {
-        // Handle any errors here
-        console.error(error);
-      });
+        .then((response) => response.json())
+        .then((data) => {
+          const pageId = data.status === 'error' ? data.currentPageId : data.nextPageId;
+          window.location.href = `/eligibility-checker/{{ meta.grantTypeId }}/${pageId}`;
+        })
+
+        .catch((error) => {
+          // Handle any errors here
+          console.error(error);
+        });
     }
   </script>
   <script>
-    document.addEventListener('DOMContentLoaded', function() {
-      const backLink = document.querySelector(".govuk-back-link");
+    document.addEventListener('DOMContentLoaded', function () {
+      const backLink = document.querySelector('.govuk-back-link');
```

Comment:
Why are we changing these back and forth? Could we stick to some IDE shared settings on these?

Reply:
Single quote is established in the prettier config, so we should move IDE settings to config settings, or discuss changing the config.